algebird-core/src/main/scala/com/twitter/algebird/Window.scala

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2018 Stripe
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.algebird
+
+import scala.collection.immutable.Queue
+import Operators._
+
+/**
+ *
+ * Convenience case class defined with a monoid for aggregating elements over
+ * a finite window.
+ *
+ * @param total Known running total of `T`
+ * @param items queue of known trailing elements.
+ *
+ *  Example usage:
+ *
+ *  case class W28[T](window: Window[T]) {
+ *    def total = this.window.total
+ *    def items = this.window.items
+ *    def size = this.window.size
+ *  }
+ *
+ *  object W28 {
+ *    val windowSize = 28
+ *    def apply[T](v: T): W28[T] = W28[T](Window(v))
+ *
+ *    implicit def w28Monoid[T](implicit p: Priority[Group[T], Monoid[T]]): Monoid[W28[T]] =
+ *      new Monoid[W28[T]] {
+ *        private val WT: Monoid[Window[T]] = WindowMonoid[T](windowSize)
+ *        def zero = W28[T](WT.zero)
+ *        def plus(a: W28[T], b: W28[T]): W28[T] =
+ *          W28[T](WT.plus(a.window, b.window))
+ *      }
+ *  }
+ *  val elements = getElements()
+ *
+ *  val trailing90Totals =
+ *    elements
+ *      .map{ W90 }
+ *      .scanLeft(W90(0)) { (a, b) => a + b }
+ *      .map{ _.total }
+ */
+
+case class Window[T](total: T, items: Queue[T]) {
+  def size = this.items.size
+}
+
+object Window {
+  def apply[T](v: T): Window[T] = Window[T](v, Queue[T](v))
+  def from[T](ts: Traversable[T])(implicit m: WindowMonoid[T]) = m.fromTraversable(ts)
```

Comment:
Can we use Iterable? Scala 2.13 is changing the collections to avoid Traversable: https://www.scala-lang.org/blog/2017/02/28/collections-rework.html#traversable-and-iterable